core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -131,68 +128,72 @@
    * @param otherFields other fields to index the set
    */
   public IndexedSet(FieldIndex<T> field, FieldIndex<T>... otherFields) {
-    mIndexMap = new HashMap<FieldIndex<T>, Integer>(otherFields.length + 1);
-    mIndexMap.put(field, 0);
+    Map<FieldIndex<T>, ConcurrentHashMap<Object, ConcurrentHashSet<T>>> indexMap =
+        new HashMap<>(otherFields.length + 1);
+    indexMap.put(field, new ConcurrentHashMap<Object, ConcurrentHashSet<T>>());
     for (int i = 1; i <= otherFields.length; i++) {
-      mIndexMap.put(otherFields[i - 1], i);
-    }
-
-    mSetIndexedByFieldValue = new ArrayList<Map<Object, Set<T>>>(mIndexMap.size());
-    for (int i = 0; i < mIndexMap.size(); i++) {
-      mSetIndexedByFieldValue.add(new HashMap<Object, Set<T>>());
+      indexMap.put(otherFields[i - 1], new ConcurrentHashMap<Object, ConcurrentHashSet<T>>());
```

Comment:
`for (int i = 0; i < otherFields.length; i++)` so that `i - 1` here is not needed